core/src/impl/Kokkos_SharedAlloc.cpp

```diff
@@ -285,50 +285,55 @@ void SharedAllocationRecord<void, void>::print_host_accessible_records(
   // allocation.
   const SharedAllocationRecord<void, void>* r = root->m_next;
 
-  char buffer[256];
-
-  if (detail) {
-    while (r != root) {
-      // Formatting dependent on sizeof(uintptr_t)
-      const char* format_string;
-
-      if (sizeof(uintptr_t) == sizeof(unsigned long)) {
-        format_string =
-            "%s addr( 0x%.12lx ) list( 0x%.12lx 0x%.12lx ) extent[ 0x%.12lx + "
-            "%.8ld ] count(%d) dealloc(0x%.12lx) %s\n";
-      } else if (sizeof(uintptr_t) == sizeof(unsigned long long)) {
-        format_string =
-            "%s addr( 0x%.12llx ) list( 0x%.12llx 0x%.12llx ) extent[ "
-            "0x%.12llx + %.8ld ] count(%d) dealloc(0x%.12llx) %s\n";
-      }
+  std::vector<char> buffer(256);
 
-      snprintf(buffer, 256, format_string, space_name,
-               reinterpret_cast<uintptr_t>(r),
-               reinterpret_cast<uintptr_t>(r->m_prev),
-               reinterpret_cast<uintptr_t>(r->m_next),
-               reinterpret_cast<uintptr_t>(r->m_alloc_ptr), r->m_alloc_size,
-               r->use_count(), reinterpret_cast<uintptr_t>(r->m_dealloc),
-               r->m_alloc_ptr->m_label);
-      s << buffer;
-      r = r->m_next;
-    }
-  } else {
-    while (r != root) {
-      // Formatting dependent on sizeof(uintptr_t)
-      const char* format_string;
-
-      if (sizeof(uintptr_t) == sizeof(unsigned long)) {
-        format_string = "%s [ 0x%.12lx + %ld ] %s\n";
-      } else if (sizeof(uintptr_t) == sizeof(unsigned long long)) {
-        format_string = "%s [ 0x%.12llx + %ld ] %s\n";
-      }
+  while (r != root) {
+    if (detail) {
+      auto printf_lambda = [&](char* buf, std::size_t buf_size) {
+        // Formatting dependent on sizeof(uintptr_t)
+        const char* format_string;
 
-      snprintf(buffer, 256, format_string, space_name,
-               reinterpret_cast<uintptr_t>(r->data()), r->size(),
-               r->m_alloc_ptr->m_label);
-      s << buffer;
-      r = r->m_next;
+        if (sizeof(uintptr_t) == sizeof(unsigned long)) {
+          format_string =
+              "%s addr( 0x%.12lx ) list( 0x%.12lx 0x%.12lx ) extent[ 0x%.12lx "
+              "+ %.8ld ] count(%d) dealloc(0x%.12lx) %s\n";
+        } else if (sizeof(uintptr_t) == sizeof(unsigned long long))
+          format_string =
+              "%s addr( 0x%.12llx ) list( 0x%.12llx 0x%.12llx ) extent[ "
+              "0x%.12llx + %.8ld ] count(%d) dealloc(0x%.12llx) %s\n";
+        return snprintf(buf, buf_size, format_string, space_name,
```

Comment:
Honestly I'd rather switch to C++ string manipulation rather than using `snprintf` here. Apparently the `-Wformat-truncation` warning can occur differently depending on whatever heuristic the compiler is using (it can even depend on optimization levels according to the gcc manual). If you think that should be done in a separate PR so as to get CI working, I'm good either way